tools/releaser/scripts/breaking-changes.sh

```diff
@@ -3,24 +3,21 @@ set -eu
 GOPATH=$(go env GOPATH)
 
 # Inputs:
-# GIT_BASE_REF - The base REF of the git repository (git rev-parse origin/main)
-# Usually "${{ github.event.pull_request.base.sha || github.event.merge_group.base_sha }}" 
 # TARGET_BREAKING_CHANGES_FILE - file to save breaking changes
 script_path=$(dirname "$0")
 
 echo "Installing go-apidiff"
 go install github.com/joelanford/go-apidiff@latest > /dev/null
 
-current_ref=$(git rev-parse main || echo)
-GIT_BASE_REF=${GIT_BASE_REF:-$current_ref}
+GIT_BASE_REF=$(git rev-parse HEAD^)
 TARGET_BREAKING_CHANGES_FILE=${TARGET_BREAKING_CHANGES_FILE:-""}
 
 echo "Running breaking changes check for $GIT_BASE_REF"
 
 pushd "$script_path/../../../" || exit ## workaround for --repo-path="../" not working
 echo "Changed directory to $(pwd)"
 set +e
-BREAKING_CHANGES=$("$GOPATH/bin/go-apidiff" "$GIT_BASE_REF" "$(git rev-parse head)" --compare-imports="false" --print-compatible="false" )
+BREAKING_CHANGES=$("$GOPATH/bin/go-apidiff" "$GIT_BASE_REF" --compare-imports="false" --print-compatible="false")
```

Comment:
Please correct my understanding:

1. CI/CD job makes 3 commits
   1. OpenAPI
   2. Code Changes
   3. Everything else/mocks/doc etc. 
2. Breaking changes engine as currently here will:
    - Compare `GIT_BASE_REF (2 - Code changes) with Current Commit (3)`
    - Should we compare `GIT_BASE_REF=(1 our base is state before code changes )  (2 - Code changes) `?

Reply:
In short. I think we should always specify 2 refs for clarity, as well as the last ref contains mock changes from golang etc. which we do not want to have